presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastoreConfig.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore;
+
+import io.airlift.units.Duration;
+
+import java.util.OptionalLong;
+
+import static java.util.Objects.requireNonNull;
+
+public class CachingHiveMetastoreConfig
+{
+    private final OptionalLong allCacheTtl;
+    private final OptionalLong allCacheRefreshInterval;
+    private final long allCacheMaxSize;
+
+    private final boolean isPartitionVersioningEnabled;
+    private final OptionalLong partitionCacheTtl;
+    private final OptionalLong partitionCacheRefreshInterval;
+    private final long partitionCacheMaxSize;
+
+    private CachingHiveMetastoreConfig(
```

Comment:
I feel this might be an overkill. Currently, we could only enable cache in an all-or-nothing fashion, and essentially we want to introduce a mode where we only enable cache for partitions. We are not really targeting a different caching strategy (ttl, refresh, size) for partitions.

Maybe we should introduce an `Enum` that is `MetastoreCacheScope` and have `ALL` and `PARTITION` in there, so that we could reuse the existing cache config.

Reply:
Yeah having a Enum will be simpler.